src/linter/block.go

```diff
@@ -64,10 +64,14 @@ type BlockWalker struct {
 	// state
 	statements map[node.Node]struct{}
 
-	// whether a function has a return without explit expression.
+	// whether a function has a return without explicit expression.
 	// Required to make a decision in void vs null type selection,
 	// since "return" is the same as "return null".
 	bareReturn bool
+	// when the function contains at least one return or yield.
+	containsReturnOrYield bool
```

Comment:
exitFlags should already contain this information in the way you want?

Reply:
I need to know if there is a `return` or `yield` in the body of the function in order to accurately determine `"mismatchingReturn"`.  I'm not sure that exitFlags holds the information I need. Comment mentions other things `// if function has exit / die / throw, then ExitFlags will be <> 0`.